src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.AspNet.Server.Kestrel.Infrastructure
 {
     public struct MemoryPoolIterator2
     {
+        private const int _maxHeaderStackLength = 16384;
```

Comment:
Total headers are generally constrained < 8k/16k So individual headers shouldn't be larger than 16k; if you are going larger than that you probably have other issues to resolve than performance - Added some related comments to: https://github.com/aspnet/KestrelHttpServer/pull/313/files#diff-6b7953b837b6ad7c921259c907520639R14

Can go larger; but that was my reasoning.

For upper levels; total stack size aspnet4 x86 was 256k, x64 was 512k; .net 1MB - not sure what the current defaults are.